### PR TITLE
Add Personal crkbd Keymap

### PR DESCRIPTION
**Pull request by toastedmangoes**
## Description

Add my personal crkbd keymap. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/waffle/combos.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#pragma once
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/combos.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/crkbd/keymaps/waffle/config.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define MASTER_LEFT
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/config.h:19]: This define isn't actually needed. It's the default behavior and can be safely removed

File: keyboards/crkbd/keymaps/waffle/config.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#    undef LOCKING_RESYNC_ENABLE
+#endif
+
+#ifndef NO_DEBUG
+#define NO_DEBUG
+#endif
+
+#if !defined(NO_PRINT) && !defined(CONSOLE_ENABLE)
+#define NO_PRINT
+#endif
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/config.h:69]: These aren't needed if console is not enabled, as they get defined automatically when console is disabled

File: keyboards/crkbd/keymaps/waffle/keycodes.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#pragma once
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/keycodes.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/crkbd/keymaps/waffle/unicodes.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/unicodes.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/crkbd/keymaps/waffle/oled.c [hunk 6]
```diff
@@ -0,0 +1,182 @@
+#pragma once
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/oled.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/crkbd/keymaps/waffle/macros.h [hunk 7]
```diff
@@ -0,0 +1,113 @@
+#pragma once
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/macros.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by toastedmangoes** [comment]
Got all the GPL2+ headers added but I totally forgot that Docker still uses avr-gcc 5.4.0 which is causing it to go over the size limit. Any advice on decreasing the size. Thanks a lot!

File: keyboards/crkbd/keymaps/waffle/oled.c [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+extern uint8_t is_master;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/waffle/oled.c:3]: Oh, heads up, this may/will be deprecated in the near future, for the "common" revision.

**Review by drashna — APPROVED** [review]

File: keyboards/crkbd/keymaps/waffle/oled.c (hunk 8 above)
Comment (toastedmangoes) [reply keyboards/crkbd/keymaps/waffle/oled.c:3]: [Split Common crkbd](https://github.com/qmk/qmk_firmware/blob/master/keyboards/crkbd/rev1/common/common.h#L24) 
Not sure if this is relevant but it looks like this split common version uses it? Or are you saying this too will be deprecated in the future? If so, what would you recommend changing about this then?

Comment (drashna) [reply keyboards/crkbd/keymaps/waffle/oled.c:3]: Foostan has a PR that changes/removes it. So ...  Worst case, hide it behind an ifdef, like the oled stuff has.

**Comment by drashna** [comment]
If Travis CI is compiling it that close, then don't worry about it (we won't).  But if you really want it under size... make sure LTO is enabled, disable features that you aren't using like the lock keys, space cadet, etc.  A lot of the core keycodes can be turned off.

File: keyboards/crkbd/keymaps/waffle/oled.c (hunk 8 above)
Comment (toastedmangoes) [reply keyboards/crkbd/keymaps/waffle/oled.c:3]: So something like 
`#ifdef SPLIT_KEYBOARD
 #    undef extern uint8_t is_master;
#endif` 
Or am I going down the wrong path here?

**Pull request closed without merging**